### PR TITLE
pkg/nodebootstrap: Fix Dropped Error

### PR DESCRIPTION
**Pull request by alrs**
This fixes a dropped `err` variable in `pkg/nodebootstrap`. Some `gofmt` indentation fixes came along for the ride.

### CODE REVIEW

File: pkg/nodebootstrap/userdata_test.go [hunk 1]
```diff
@@ -49,14 +49,14 @@ var _ = Describe("User data", func() {
 		It("contains default kube reservations", func() {
 			ng.InstanceType = "i3.metal"
 			data, err := makeKubeletConfigYAML(clusterConfig, ng)
-
+			Expect(err).ToNot(HaveOccurred())
```
Comment (sayboras) [inline pkg/nodebootstrap/userdata_test.go:52]: Good spot :100:

**Comment by sayboras** [comment]
There is TODO item, which will enable linting rule for test files as well. I supporsed to do it, but forgot after a long time.

All we need to do is to make small changes in .golang.yaml, and run `make lint`. Pleaset shout out if you have some time for it, otherwise I will do it later in seperate PR.

```
diff --git a/.golangci.yml b/.golangci.yml
index ccd3d8ec..b74e771e 100644
--- a/.golangci.yml
+++ b/.golangci.yml
@@ -10,11 +10,12 @@ run:
   issues-exit-code: 1
 
   # include test files or not, default is true
-  tests: false
+  tests: true
 
   # list of build tags, all linters use it. Default is empty list.
-  #build-tags:
-  #  - mytag
+  build-tags:
+    - release
+    - integration
 ```

<details>
<summary>Current linting errors, your changes help to fix a few as well</summay>

```
$ make lint
time "/home/tammach/go/bin/golangci-lint" run
pkg/ctl/set/set_test.go:48:2                                                   unused       field `cmd` is unused
pkg/vpc/vpc_test.go:54:2                                                       unused       field `describeSubnetsOutput` is unused
pkg/ctl/delete/delete_test.go:48:2                                             unused       field `cmd` is unused
pkg/ctl/generate/generate_test.go:47:2                                         unused       field `cmd` is unused
pkg/nodebootstrap/userdata_test.go:51:10                                       ineffassign  ineffectual assignment to `err`
pkg/iam/oidc/api_test.go:76:16                                                 errcheck     Error return value of `srv.serve` is not checked
pkg/iam/oidc/api_test.go:97:16                                                 errcheck     Error return value of `srv.serve` is not checked
pkg/iam/oidc/api_test.go:176:16                                                errcheck     Error return value of `srv.serve` is not checked
pkg/gitops/gitops_test.go:72:16                                                errcheck     Error return value of `io.RemoveAll` is not checked
pkg/gitops/gitops_test.go:73:16                                                errcheck     Error return value of `io.RemoveAll` is not checked
pkg/gitops/gitops_test.go:181:12                                               errcheck     Error return value of `createFile` is not checked
pkg/gitops/gitops_test.go:182:12                                               errcheck     Error return value of `createFile` is not checked
pkg/gitops/gitops_test.go:183:12                                               errcheck     Error return value of `createFile` is not checked
pkg/gitops/gitops_test.go:185:13                                               errcheck     Error return value of `memFs.Mkdir` is not checked
pkg/ctl/cmdutils/nodegroup_filter_test.go:56:29                                errcheck     Error return value of `filter.AppendExcludeGlobs` is not checked
pkg/ctl/cmdutils/nodegroup_filter_test.go:99:29                                errcheck     Error return value of `filter.AppendExcludeGlobs` is not checked
pkg/ctl/cmdutils/nodegroup_filter_test.go:114:29                               errcheck     Error return value of `filter.AppendExcludeGlobs` is not checked
pkg/ctl/cmdutils/nodegroup_filter_test.go:149:18                               errcheck     Error return value of `filter.ForEach` is not checked
pkg/ctl/cmdutils/nodegroup_filter_test.go:156:18                               errcheck     Error return value of `filter.ForEach` is not checked
pkg/ctl/cmdutils/nodegroup_filter_test.go:165:20                               errcheck     Error return value of `printer.PrintObj` is not checked
pkg/ctl/cmdutils/nodegroup_filter_test.go:178:18                               errcheck     Error return value of `filter.ForEach` is not checked
pkg/utils/names/names_test.go:3                                                goimports    File is not `goimports`-ed
pkg/nodebootstrap/userdata_test.go:58                                          goimports    File is not `goimports`-ed
pkg/vpc/vpc_test.go:626                                                        goimports    File is not `goimports`-ed
integration/tests/params.go:12:2                                               golint       should not use dot imports
integration/tests/quickstart_profiles_gen/quickstart_profiles_gen_test.go:3:1  golint       don't use an underscore in package name
integration/tests/backwards_compat/backwards_compatibility_test.go:3:1         golint       don't use an underscore in package name
pkg/cfn/builder/api_test.go:76:2                                               golint       struct field `CidrIp` should be `CidrIP`
pkg/cfn/builder/api_test.go:79:2                                               golint       struct field `VpcId` should be `VpcID`
pkg/cfn/builder/api_test.go:80:2                                               golint       struct field `RouteTableId` should be `RouteTableID`
pkg/cfn/builder/api_test.go:81:2                                               golint       struct field `GatewayId` should be `GatewayID`
pkg/cfn/builder/api_test.go:117:26                                             golint       struct field `ImageId` should be `ImageID`
pkg/cfn/builder/api_test.go:122:3                                              golint       struct field `AssociatePublicIpAddress` should be `AssociatePublicIPAddress`
pkg/version/generate/release_generate.go:13:2                                  golint       should not use dot imports
pkg/version/generate/release_generate.go:19:7                                  golint       const `defaultPreReleaseId` should be `defaultPreReleaseID`
pkg/version/generate/release_generate.go:73:2                                  golint       should replace `ver.Minor += 1` with `ver.Minor++`
pkg/version/generate/release_generate.go:77:34                                 golint       func parameter `preReleaseId` should be `preReleaseID`
pkg/fargate/client_test.go:216:26                                              golint       error strings should not be capitalized or end with punctuation or a newline
pkg/fargate/client_test.go:367:28                                              golint       error strings should not be capitalized or end with punctuation or a newline
pkg/fargate/client_test.go:375:28                                              golint       error strings should not be capitalized or end with punctuation or a newline
integration/matchers/flux.go:12:2                                              golint       should not use dot imports
integration/tests/cluster_api/cluster_api_endpoints_test.go:3:1                golint       don't use an underscore in package name
pkg/ami/auto_resolver_test.go:16:2                                             golint       struct field `imageId` should be `imageID`
pkg/ami/auto_resolver_test.go:229:86                                           golint       func parameter `amiId` should be `amiID`
pkg/ami/ssm_resolver_test.go:284:69                                            golint       func parameter `amiId` should be `amiID`
integration/tests/crud/creategetdelete_test.go:541:8                           golint       struct field `AssumedRoleId` should be `AssumedRoleID`
integration/tests/crud/creategetdelete_test.go:545:51                          golint       struct field `AccessKeyId` should be `AccessKeyID`
pkg/eks/api_test.go:198:55                                                     golint       func parameter `amiId` should be `amiID`
integration/matchers/flux.go:7                                                 gofmt        File is not `gofmt`-ed
integration/main.go:26:42                                                      staticcheck  SA1016: os.Kill cannot be trapped (did you mean syscall.SIGTERM?)
pkg/authconfigmap/authconfigmap_test.go:346:4                                  staticcheck  SA4006: this value of `cm` is never used
pkg/authconfigmap/authconfigmap_test.go:352:4                                  staticcheck  SA4006: this value of `cm` is never used
pkg/authconfigmap/authconfigmap_test.go:358:4                                  staticcheck  SA4006: this value of `cm` is never used
integration/tests/cluster_api/cluster_api_endpoints_test.go:144:7              gosimple     S1039: unnecessary use of fmt.Sprintf
pkg/cfn/builder/api_matchers_test.go:48:9                                      gosimple     S1039: unnecessary use of fmt.Sprintf
integration/utilities/kube/kube.go:13:35                                       gosimple     S1023: redundant `return` statement
```

</details>

**Comment by alrs** [comment]
@sayboras I added your `.golangci.yml` changes to this PR.

**Comment by sayboras** [comment]
> @sayboras I added your `.golangci.yml` changes to this PR.

Thanks, there is a list of linting issues that I shared in previous comment as well.

File: pkg/testutils/client.go [hunk 2]
```diff
@@ -6,7 +6,7 @@ import (
 	"io/ioutil"
 	"net/http"
 
-	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega"
```
Comment (sayboras) [inline pkg/testutils/client.go:9]: for this one, i think we can exclude the rules like the below

```
issues:
  # Excluding configuration per-path, per-linter, per-text and per-source
  exclude-rules:
    # Exclude some staticcheck messages
    - linters:
        - staticcheck
      text: "SA1016:"
    # Exclude some golint messages
    - linters:
        - golint
      text: "should not use dot imports"
    - linters:
        - golint
      text: "don't use an underscore in package name"
```

File: pkg/utils/kubectl/kubectl.go [hunk 3]
```diff
@@ -89,7 +89,7 @@ func CheckAllCommands(kubeconfigPath string, isContextSet bool, contextName stri
 			return errors.Wrapf(err, "parsing Kubernetes version string %q return by the EKS API server", version)
 		}
 		if version.Major == 1 && version.Minor < 10 {
-			return fmt.Errorf("Kubernetes version %s found, v1.10.0 or newer is expected with EKS %s", serverVersion, suggestion)
+			return fmt.Errorf("kubernetes version %s found, v1.10.0 or newer is expected with EKS %s", serverVersion, suggestion)
```
Comment (sayboras) [inline pkg/utils/kubectl/kubectl.go:92]: just curious why this one is required, as the daily build seems passing for this file.

File: pkg/authconfigmap/authconfigmap_test.go [hunk 4]
```diff
@@ -390,20 +390,20 @@ var _ = Describe("AuthConfigMap{}", func() {
 		}
 
 		It("should remove role and user", func() {
-			cm := removeAndSave(roleB)
-			cm = removeAndSave(userB)
+			_ = removeAndSave(roleB)
```
Comment (sayboras) [inline pkg/authconfigmap/authconfigmap_test.go:393]: we can still keep this like, and move the first Assertion `Expect(cm.Data["mapRoles"]).To(MatchYAML(expectedRoleA + expectedRoleA))` right after this line.

File: pkg/testutils/client.go (hunk 2 above)
Comment (alrs) [reply pkg/testutils/client.go:9]: Got it.

File: pkg/utils/kubectl/kubectl.go (hunk 3 above)
Comment (alrs) [reply pkg/utils/kubectl/kubectl.go:92]: I was using `staticcheck` directly, apparently `golang-ci` only uses a subset of its lint reports, I"ll remove these.

**Comment by alrs** [comment]
I've temporarily pulled out `.golangci.yml` changes so that this PR can land as-is without becoming a merge-conflict nightmare over time. I'll keep going in subsequent pull requests.

**Comment by sayboras** [comment]
> I've temporarily pulled out `.golangci.yml` changes so that this PR can land as-is without becoming a merge-conflict nightmare over time. I'll keep going in subsequent pull requests.

No worry. Then I think you can _rebase_ with upstream master branch, and keep only your original first commit in this PR. As without changes in .golangci.yaml, it's not clear which change is related to which linting rule.

You can keep a copy of current changes you already made and send them another PR :)

**Comment by alrs** [comment]
OK, back to the original bugfix, plus `go fmt`. I'd like for the `go fmt` fixes to land and not be a source of merge conflicts later.

File: integration/matchers/flux.go [hunk 5]
```diff
@@ -4,18 +4,18 @@ package matchers
...
 import (
 	"fmt"
-	"io/ioutil"
-	"os"
-	"os/exec"
-	"path/filepath"
-	"github.com/weaveworks/eksctl/integration/utilities/git"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/weaveworks/eksctl/integration/utilities/git"
```
Comment (sayboras) [inline integration/matchers/flux.go:9]: seems like you are sorting imports, can you run `goimports -w <file path>` for this file instead ?

**Comment by sayboras** [comment]
Apart from one minor comments for `goimports`, it's LGTM :tada:

@martina-if @cPu1 @marccarre For your input and reviews :pray:

File: integration/matchers/flux.go (hunk 5 above)
Comment (alrs) [reply integration/matchers/flux.go:14]: fixed, and `--force-push`ed.

**Comment by martina-if** [comment]
Hi @alrs thank you for this PR! Was the initial intent of the PR to fix pkg/nodebootstrap/userdata_test.go or pkg/nodebootstrap/userdata.go? I'm asking because it looks like the initial change was pulled out? Or I can't see it at least.

**Comment by alrs** [comment]
> Hi @alrs thank you for this PR! Was the initial intent of the PR to fix pkg/nodebootstrap/userdata_test.go or pkg/nodebootstrap/userdata.go? I'm asking because it looks like the initial change was pulled out? Or I can't see it at least.

Yeah, the fix is at 923b070f86faad1dbf9400d08feb7baf5c2ebcbd, in `pkg/nodebootstrap/userdata_test.go`

**Pull request merged**